packages/carbon-web-components/src/components/side-panel/side-panel-header.ts

```diff
@@ -0,0 +1,276 @@
+
+  // /**
+  //  * has test this
+  //  */
+  // @state() _hasTestThis;
+
+  // _handleSlotChange(e: Event) {
+  //   // can change element/node attributes here
+  //   const target = e.target as HTMLSlotElement;
+  //   const nodes = target.assignedElements();
+  //   nodes[0].setAttribute('style', 'color: red;');
+  //   console.log(nodes.length);
+
+  //   this._hasTestThis = nodes.length > 0;
+  // }
+
+  // updated() {
+  //   // can see slotted children here
+  //   console.log(this.children);
+  // }
+
+  // eslint-disable-next-line class-methods-use-this
+  _handleActionToolbarChange(e: Event) {
+    const target = e.target as HTMLSlotElement;
+    const assigned = target?.assignedElements();
+    const actionSlot = assigned?.[0];
+    const actions = actionSlot?.children;
+
+    if (actions) {
+      for (let action of actions) {
+        // action size should always be md
+        action.setAttribute('size', 'md');
+      }
+    }
+  }
+
+  // createRenderRoot() {
+  //   // Do not drill down into a new shadow dom.
+  //   return this;
+  // }
+
+  render() {
+    const {
+      animateTitle,
+      closeIconDescription,
+      currentStep,
+      labelText,
+      navigationBackIconDescription,
+      // slug,
+      subtitle,
+      title,
+    } = this;
+    const blockClass = `${prefix}--side-panel`;
+
+    // let normalizedSlug;
+    // if (slug) {
+    //   normalizedSlug = React.cloneElement(slug, {
+    //     // slug size is sm unless actions and size > md
+    //     size: actions.length && /l/.test(size) ? 'md' : 'sm',
+    //   });
+    // }
+
+    const reducedMotion =
+      typeof window !== 'undefined' && window?.matchMedia
+        ? window.matchMedia('(prefers-reduced-motion: reduce)')
+        : { matches: true };
+
+    const titleContainerClasses = classMap({
+      [`${blockClass}__title-container`]: true,
+      [`${blockClass}__on-detail-step`]: currentStep > 0,
+      [`${blockClass}__on-detail-step-without-title`]:
+        currentStep > 0 && !title,
+      [`${blockClass}__title-container--no-title-animation`]: !animateTitle,
+      [`${blockClass}__title-container-without-title`]: !title,
+      [`${blockClass}__title-container--reduced-motion`]: reducedMotion.matches,
+    });
+
+    const subtitleClasses = classMap({
+      [`${blockClass}__subtitle-text`]: true,
+      [`${blockClass}__subtitle-text-no-animation`]: !animateTitle,
+      // [`${blockClass}__subtitle-text-no-animation-no-action-toolbar`]:
+      //   !animateTitle &&
+      //   (!actionToolbarButtons || !actionToolbarButtons.length),
+      // [`${blockClass}__subtitle-text-is-animating`]:
+      //   !animationComplete && animateTitle,
+      [`${blockClass}__subtitle-without-title`]: !title,
+    });
+
+    //  const titleEl = title && title.length && labelText && labelText.length ?
+    //   html`<p class={`${prefix}__label-text`}>{labelText}</p>` : html``;
+
+    return html`
+      <div class=${titleContainerClasses}>
```

Comment:
we could do what Ari suggested here, where we check the attributes passed into the components in the scss file and display accordingly by extending the classes

Reply:
Done